File: Foundation/Module 2/Day 1/11.c
#include <stdio.h>

int main()
{
    int myArray[7] = {25, 12, 8, 45, 3, 67, 19};
    int smallestNum = myArray[0];

    for (int i = 0; i < 7; i++)
    {
        if (smallestNum > myArray[i])
        {
            smallestNum = myArray[i];
        }
    }

    printf("The Smallest Number in the Array: %d", smallestNum);
}

File: FromHub/Module 7/Assignments/11.js
/*
Write a function to find the second-largest number in an array.
E.g.: Input → [10, 5, 20, 8, 25]
Output → 20
*/

function secondLargest(array) {
  let largest = 0;
  let secLargest = 0;

  for (let num of array) {
    if (num > largest) {
      secLargest = largest; // update second largest before changing largest
      largest = num;
    } else if (num > secLargest && num < largest) {
      secLargest = num;
    }
  }

  return secLargest;
}

let myArray = [10, 5, 20, 8, 25];
console.log(secondLargest(myArray));
